File: server/src/main/java/com/bookmyvenue/server/payment/controller/PaymentController.java
package com.bookmyvenue.server.payment.controller;

import com.bookmyvenue.server.payment.dto.request.VerifyPaymentRequest;
import com.bookmyvenue.server.payment.dto.response.PaymentResponse;
import com.bookmyvenue.server.payment.service.PaymentService;
import com.razorpay.RazorpayException;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/v1/payments")
@Tag(name = "Payment Management")
@PreAuthorize("hasRole('USER')")
public class PaymentController {

    private final PaymentService paymentService;

    @PostMapping("/{bookingId}")
    @Operation(summary = "Create Payment Order")
    public ResponseEntity<PaymentResponse> createPayment(@PathVariable Long bookingId) throws RazorpayException {

        return ResponseEntity.ok(
                paymentService.createPayment(bookingId));
    }


    @PostMapping("/verify")
    @Operation(summary = "Verify Payment")
    public ResponseEntity<Void> verifyPayment(@RequestBody VerifyPaymentRequest request) throws RazorpayException {

        paymentService.verifyPayment(request);
        return ResponseEntity.ok().build();
    }
}

File: bmv-main/src/payment/payment.controller.ts
import {
  Body,
  Controller,
  Post,
  Req,
  UseGuards,
} from '@nestjs/common';
import {
  ApiBearerAuth,
  ApiOperation,
  ApiTags,
} from '@nestjs/swagger';
import { VerifyPaymentDto } from './dto/verify-order.dto';
import { PaymentService } from './payment.service';
import { CreateOrderDto } from './dto/create-order.dto';
import { JwtAuthGuard } from '../guard/jwt.guard';

@ApiTags('Payment')
@ApiBearerAuth()
@UseGuards(JwtAuthGuard)
@Controller('payments')
export class PaymentController {
  constructor(
    private readonly paymentService: PaymentService,
  ) {}

  @Post('create-order')
  @ApiOperation({
    summary: 'Create Razorpay order',
  })
  createOrder(
    @Body() createOrderDto: CreateOrderDto,
    @Req() req,
  ) {
    return this.paymentService.createOrder(
      createOrderDto,
      req.user.userId,
    );
  }
  @Post('verify')
@ApiOperation({
  summary: 'Verify Razorpay payment',
})
verifyPayment(
  @Body() verifyPaymentDto: VerifyPaymentDto,
  @Req() req,
) {
  return this.paymentService.verifyPayment(
    verifyPaymentDto,
    req.user.userId,
  );
}
}
